main.go

```diff
@@ -495,6 +512,31 @@ func main() {
 		}
 	}
 
+	if *flGithubAppPrivateKeyFile != "" {
+		if *flGithubAppApplicationID == 0 {
+			handleConfigError(log, true, "ERROR: --github-app-application-id must be specified when --github-app-private-key-file is specified")
+		}
+		if *flGithubAppInstallationID == 0 {
+			handleConfigError(log, true, "ERROR: --github-app-installation-id must be specified when --github-app-private-key-file is specified")
+		}
+		if *flUsername != "" {
+			handleConfigError(log, true, "ERROR: --username may not be specified when --github-app-private-key-file is specified")
+		}
+		if *flPassword != "" {
+			handleConfigError(log, true, "ERROR: --password may not be specified when --github-app-private-key-file is specified")
+		}
+		if *flPasswordFile != "" {
+			handleConfigError(log, true, "ERROR: --password-file may not be specified when --github-app-private-key-file is specified")
+		}
+	} else {
+		if *flGithubAppApplicationID != 0 {
+			handleConfigError(log, true, "ERROR: --github-app-application-id may only be specified when --github-app-private-key-file is specified")
+		}
+		if *flGithubAppInstallationID != 0 {
+			handleConfigError(log, true, "ERROR: --github-app-installation-id may only be specified when --github-app-private-key-file is specified")
+		}
+	}
```

Comment:
This makes the Github app auth mutually exclusive with username/password. What about other auth types, like ssh,  cookiefile, or etc?

Reply:
I think I omitted those because I didn't see them being checked in the existing validations, but I can add them sure

Reply:
Yeah, this is a mess.  Was OK when it was 3 mutually exclusive modes, but it's getting out of hand.  OK for now, but we should plan a followup which is more declarative feeling and handles the mutual exclusion more automatically.